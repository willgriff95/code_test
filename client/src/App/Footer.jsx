import React from 'react'
import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableFooter from '@material-ui/core/TableFooter';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';

const styles = theme => ({

  })

const Footer = ({classes, rowsPerPage, page, data, handleChangePage, handleChangeRowsPerPage}) => {

    return(
    <TableFooter className={classes.pageNav}>
        <TableRow>
        <TableCell>
            <TablePagination
                labelRowsPerPage={''}
                rowsPerPageOptions={[]}
                component="div"
                count={data.length}
                rowsPerPage={rowsPerPage}
                page={page}
                backIconButtonProps={{
                    'aria-label': 'Previous Page',
                }}
                nextIconButtonProps={{
                    'aria-label': 'Next Page',
                }}
                onChangePage={handleChangePage}
                onChangeRowsPerPage={handleChangeRowsPerPage}
            />
        </TableCell>
        </TableRow>
    </TableFooter>
    )
}

export default withStyles(styles)(Footer)