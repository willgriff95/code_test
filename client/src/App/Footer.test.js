import React from 'react';
import ReactDOM from 'react-dom';
import Footer from './Footer';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer />, div);
  ReactDOM.unmountComponentAtNode(div);
})

// Check component is receiving the following props
// rowsPerPage, page, data, handleChangePage, handleChangeRowsPerPage


it('render correctly Footer component', () => {  
    const FooterComponent = renderer.create(<Footer />).toJSON();
    expect(FooterComponent).toMatchSnapshot();
});