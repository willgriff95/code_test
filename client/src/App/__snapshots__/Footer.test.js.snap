// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Footer component 1`] = `
<tfoot
  className="MuiTableFooter-root-94"
>
  <tr
    className="MuiTableRow-root-95 MuiTableRow-footer-99"
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root-100 MuiTableCell-footer-103"
    >
      <div
        className="MuiTablePagination-root-112"
      >
        <div
          className="MuiToolbar-root-122 MuiToolbar-regular-124 MuiToolbar-gutters-123 MuiTablePagination-toolbar-113"
        >
          <div
            className="MuiTablePagination-spacer-114"
          />
          <span
            className="MuiTypography-root-126 MuiTypography-caption-136 MuiTypography-colorInherit-155 MuiTablePagination-caption-115"
          >
            0-NaN of 0
          </span>
          <div
            className="MuiTablePagination-actions-121"
          >
            <button
              aria-label="Previous Page"
              className="MuiButtonBase-root-168 MuiIconButton-root-162 MuiIconButton-colorInherit-163"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-167"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-171"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-180"
              />
            </button>
            <button
              aria-label="Next Page"
              className="MuiButtonBase-root-168 MuiIconButton-root-162 MuiIconButton-colorInherit-163"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-167"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-171"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-180"
              />
            </button>
          </div>
        </div>
      </div>
    </td>
  </tr>
</tfoot>
`;
